python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -262,6 +263,33 @@ def _get_ray_resources_from_group_spec(
     if num_gpus is not None:
         resources["GPU"] = num_gpus
 
+    if num_tpus is not None:
+        # Add TPU Ray resource if not already added by ray_start_params,
+        # but specified in k8s_resource_limits.
+        if "TPU" not in custom_resource_dict:
```

Comment:
Do we need to check whether `TPU` is in `custom_resource_dict` here? We have already checked `custom_resource_dict` in `_get_num_tpus`.

Reply:
Removed this logic in be3af8e, we can add back in the auto-detection of the TPU resource based on the k8s resource limit on the Kuberay side for consistency in a follow-up PR.